File: adventofcode/year2015/day01.py
from adventofcode.common import Solution


class Day01(Solution):
    def __init__(self, year: str, day: str):
        super().__init__(year, day)

        self._input = self._load_input_as_string()

    def part_one(self):
        return self._input.count('(') - self._input.count(')')

    def part_two(self):
        floor = 0
        position = 0
        while floor >= 0:
            floor += 1 if self._input[position] == '(' else -1
            position += 1

        return position

File: adventofcode/year2017/Day01.cs
using adventofcode.common;

namespace adventofcode.year2017;

public class Day01 : Solution
{
    private string _input;
    
    public Day01(string year, string day) : base(year, day)
    {
        _input = LoadInputAsString();
    }

    public override string PartOne()
    {
        var sum = 0;
        for (int i = 0; i < _input.Length; i++)
        {
            if (i + 1 < _input.Length)
            {
                sum += _input[i] == _input[i + 1] ? Convert.ToInt32(_input[i].ToString()) : 0;
            }
            else
            {
                sum += _input[i] == _input[0] ? Convert.ToInt32(_input[i].ToString()) : 0;
            }
        }

        return Convert.ToString(sum);
    }

    public override string PartTwo()
    {
        var offset = _input.Length / 2;
        var sum = 0;
        for (int i = 0; i < _input.Length; i++)
        {
            if (i + offset < _input.Length)
            {
                sum += _input[i] == _input[i + offset] ? Convert.ToInt32(_input[i].ToString()) : 0;
            }
            else
            {
                sum += _input[i] == _input[(i + offset) % _input.Length] ? Convert.ToInt32(_input[i].ToString()) : 0;
            }
        }

        return Convert.ToString(sum);
    }
}